src/uct/ib/mlx5/ib_mlx5.h

```diff
@@ -45,7 +45,7 @@
 #else
 
 #  define mlx5_av_base(_av)         (_av)
-#  define mlx5_av_grh(_av)          (_av)
+#  define mlx5_av_grh(_av)          ((struct mlx5_grh_av *)(&(_av)->reserved0[0]))
```

Comment:
@yosefe could you look at this?
it seems original macro was not correct: in ud_mlx5_common.c:64 there is code:
```
if (*is_global) {
    ucs_assert_always(grh_av != NULL);
    memcpy(grh_av, mlx5_av_grh(&mlx5_av), sizeof(*grh_av));
}
```
which will fail (will totally break mlx5_av object)

Reply:
yep, the original (invalid) assumption was that the user of that macro will only try to access the fields and not the struct pointer. i guess it was never tested with older OFED versions. on which setup did you see this fail?

Reply:
There were issued in compilation, i just looked how this structure is used, and found that it could cause problems